Dockerfile

```diff
@@ -1,11 +1,17 @@
-FROM debian:buster
+FROM debian:buster-20190910
```

Comment:
Why do you want to pin this?  It will just add more work to keep moving this up to date in future (which will probably end up having to be done by me)

Reply:
@Page-  Pining this, requires an additional step when updating the base image, but adds control and transparency in the process. We need a more explicit version control here, otherwise, we 'd have to figure out which buster version was out at build time, etc in a future issue.

Reply:
With a floating base image, we could end up with a different  image, depending on the time of the build, despite using the same codebase and Dockerfile.

Reply:
So, I'm a bit confused as to the 'we could end up with a different image' argument. To what? Once we've built an image for `open-balena-base` and merge it, we've got a versioned base image that is used by downstream projects. If it gets rebuilt with a new version of Buster, then that by the very way we do this (merging to `master`) will be a different versioned image. I'm a bit confused here. And as @Page- says, given the fact that Buster's updated every month with security updates, I don't see any advantage to pinning this. We have a script to update this, and test all downstream projects. The only thing I can think of is that there's a difference between when the PR goes up and is approved and when the merge occurs... but the way balenaCI works means it'll use the previously cached image that was last built anyway, won't it?

Could you clarify the issue a bit more?

Reply:
We also have an explicit
```
RUN apt-get update \
	&& apt-get dist-upgrade \
```
below that will kind of nullify this pinning if that gets rerun, at the cost that it will increase the total image size vs just using an updated base image

Reply:
hey @hedss a floating base image means that the Dockerfile is not declarative with regards to the version of the base. What if balenaCI runs a built around the time a new buster base is published? What if someone hits rebuild? We can infer the base image used by comparing layers, or getting datetime of the build + the builder cache involved for the apt operations, but an alternative would just involve using declarative references. Updating the script for tracking the current buster base image used would rule out build time from this. On the other hand, we can treat the product image as a black box, and rely on the downstream component tests.
Page- good catch! Shall we remove dist-upgrade and rely on the base image for upgrades (or at least replace it with upgrade for some predictability)? That would apply both on the floating and pinned base image tags.

Reply:
@mikesimos I'm aware the base image may change from version to version of `open-balena-base` (and as Pagan's pointed out, that gets nullified anyway), but we currently implicitly understand that a new version of `open-balena-base` may do that (and include, for example, any updates and security changes that have recently been added). I'd be more in agreement if it weren't for the fact that the base images we're inheriting from are tested thoroughly by Debian before being released, and if we'd ever seen an issue.

I have *way* more of a worry that we'll start to see issues because `open-balena-base` won't get updated regularly. And I worry this because I had to deal with exactly this issue when I started BoB, finding that almost every component bar the API frequently lagged behind updating the base, in one case by a year. So we either pin and ensure anytime a new base release occurs we audit it and determine whether we update, or we trust Debian. I don't care which, but either way it needs a process.